test/typescript/dispatch.ts

```diff
@@ -1,12 +1,15 @@
 import {Dispatch, Action} from "../../index.d.ts";
 
 
-declare const dispatch: Dispatch;
-
+declare const dispatch: Dispatch<any>;
 
 const dispatchResult: Action = dispatch({type: 'TYPE'});
 
+// thunk
+declare module "../../index.d.ts" {
+    export interface Dispatch<S> {
```

Comment:
Can't `S` be inferred as well? I wouldn't add it directly as a parameter to `Dispatch`, because it's specific to Thunk. Other middlewares will have their own parameters too.